### PR TITLE
Containerize ZK for ECS

### PR DESCRIPTION
**Pull request by vicaraniron**
EFS is created first outside the stack

ECS and ZK are created by the master_template (a nested stack)

### CODE REVIEW

File: master-template.yaml [hunk 1]
```diff
@@ -2,18 +2,17 @@
...
 
-  Finally, it deploys a pair of example ECS services from containers published in
-  Amazon EC2 Container Registry (Amazon ECR).
+  It then deploys a Zookeeper cluster from containers published in Docker Hub.
+
+  TODO: move containers to ECR?
 
 Parameters:
+  EfsStackName:
+    Description: Key Pair name for SSH access
```
Comment (dloiko) [inline master-template.yaml]: Description for this parameter appears to be incorrect - maybe "Name of the deployed cloudformation stack for EFS"

File: zookeeper/zk-cluster.yaml [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+AWSTemplateFormatVersion: '2010-09-09'
+
+Description: Zookeeper Cluster
+
+Parameters:
+  Cluster:
+    Description: ECS Cluster
+    Type: String
+  StackName:
+    Description: Stack Name
```
Comment (dloiko) [inline zookeeper/zk-cluster.yaml]: Which stack is this referring to? Looking at the parameters passed it, it seems like the master one, but not sure. Maybe a clearer description here, or a different parameter name would help.

File: deploy_stack [hunk 3]
```diff
@@ -4,21 +4,32 @@ set -o errexit
 set -o pipefail
 [[ "${DEBUG}" == 'true' ]] && set -o xtrace
 
-# TODO loop over templates
+readonly CFNS=$(grep --exclude=deploy_stack --exclude-dir={kafka-connect,kafka-rest,schema-registry} -rnwl . -e 'AWSTemplateFormatVersion')
```
Comment (hackmad) [inline deploy_stack]: `readonly`: learnt something new in bash 👍

File: deploy_stack [hunk 4]
```diff
@@ -4,21 +4,32 @@ set -o errexit
...
 
-ERRORS=$(aws cloudformation validate-template --template-body file://master-template.yaml | jq .Error)
+for cfn in ${CFNS}; do
+    echo "Validating template ${cfn}"
+    ERRORS=$(aws cloudformation validate-template --template-body file://${cfn} | jq .Error)
```
Comment (hackmad) [inline deploy_stack]: What if `jq` is not present on the system. This thing will ignore the next line and try to update the stack.

File: zookeeper/zk-service.yaml [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+AWSTemplateFormatVersion: '2010-09-09'
+
+Description: Creates an ECS Zookeeper Service and Task
+
+Parameters:
+  Cluster:
+    Description: ECS Cluster
+    Type: String
+  StackName:
+    Description: Stack Name
```
Comment (dloiko) [inline zookeeper/zk-service.yaml]: Same as other stack name - not sure what it's referring to

**Comment by hackmad** [comment]
@dloiko I updated the parameter descriptions

**Pull request merged**